lib/management.js

```diff
@@ -245,64 +297,72 @@ function saveServiceCredentials(conf, auth) {
     // initialized = true;
 }
 
-function loadOverlayVersion(db, version, cb) {
-    db.get(`configuration/overlay/${version}`, {}, (err, val) => {
+function loadOverlayVersion(metadata, version, cb) {
+    metadata.getObjectMD(serviceBucket, `configuration/overlay/${version}`, {},
+    logger, (err, val) => {
         if (err) {
             return cb(err);
         }
-        return cb(null, JSON.parse(val));
+        return cb(null, val);
     });
 }
 
 function initManagement(done) {
-    // if (initialized) {
-    //     return process.nextTick(done);
-    // }
-
-    const mdDb = mdClient.openDB(error => {
-        if (error) {
-            return process.nextTick(done, error);
-        }
-
-        const db = mdDb.openSub('PENSIEVE');
+    let setup = false;
+    function iterate(done) {
+        return async.waterfall([
+            cb => {
+                if (!setup) {
+                    setup = true;
+                    metadata = new Metadata('mongodb', params, bucketclient,
+                        logger);
+                    return metadata.setup(() => cb());
+                }
+                return process.nextTick(cb);
+            },
+            cb => metadata.listObject(serviceBucket, {}, logger, () => cb()),
+            cb => {
+                metadata.getObjectMD(serviceBucket,
+                'configuration/overlay-version', {}, logger, (err, res) =>
+                    cb(err, res));
+            },
+            (version, cb) => {
```

Comment:
could be written as `(version, cb) => loadOverlayVersion(metadata, version, cb)`, as this waterfall is becoming busy